### PR TITLE
Same as Solve the NPE in Issue #202, but rebased

### PR DESCRIPTION
**Pull request by dcbriccetti**
I rebased on @mdudzinski’s changes to undo that the cherry picking I did before, in https://github.com/opendatakit/javarosa/pull/203, put my name on his changes.

### CODE REVIEW

File: test/org/javarosa/xform/parse/ChildProcessingTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Test public void worksWithTwoNotMatchingChildren() {
+        Element el = new Element();
+        el.addChild(ELEMENT, el.createElement(null, "Child Name 1"));
+        el.addChild(ELEMENT, el.createElement(null, "Child Name 2"));
+        assertFalse(XFormParser.allChildrenAreElementsAndHaveTheSameName(el));
+    }
+
+    @Test public void WorksWithNoChildren() {
```
Comment (mdudzinski) [inline test/org/javarosa/xform/parse/ChildProcessingTest.java]: Should start with lowercase.

**Comment by lognaturel** [comment]
#203 has been merged but I can also merge this one to get the right authorship in there (though @dcbriccetti will appear to have previously made the same changes @mdudzinski made).

Good catch on the case issue, @mdudzinski! Could you please fix that real quick, @dcbriccetti?

**Comment by dcbriccetti** [comment]
Hmm, why a conflict? @lognaturel, can you force that in or whatever the needful is?

**Comment by mdudzinski** [comment]
@dcbriccetti @lognaturel  Shouldn't such typos be caught by checkstyle?

**Comment by lognaturel** [comment]
@mdudzinski Yes but that rule must not be turned on yet. My guess is that the code base is pretty inconsistent so it may take some work to turn on. I think there's a way to turn on checks only for commits after a certain point in time and perhaps we should look into that.

Sorry, I'm not sure what to do with this! I can fix the merge conflict and then merge with a merge commit but that would make the history really strange.

**Comment by dcbriccetti** [comment]
OK, let’s abandon this. I’ll fix the WorksWithNoChildren in the next PR in this area.

**Comment by lognaturel** [comment]
🙏

**Comment by dcbriccetti** [comment]
A slight load off.

**Comment by lognaturel** [comment]
Every little bit counts.

**Pull request closed without merging**